lib/common/api/deprecate.js

```diff
@@ -46,6 +46,26 @@ deprecate.log = (message) => {
   }
 }
 
+// Deprecate an event.
+deprecate.event = (emitter, oldName, newName) => {
+  let warned = false
+  return emitter.on(newName, function (...args) {
+    // There are no listeners for this event
+    if (this.listenerCount(oldName) === 0) { return }
+    // noDeprecation set or if user has already been warned
+    if (warned || process.noDeprecation) { return }
+    warned = true
+    // Event name does not start with a -, so it's not an internal event.
```

Comment:
I didn't mean to criticize this piece too hard, but now this line just sounds too complicated.
I _think_ that it can become clearer if it is rewritten as something like this:

```js
const isInternalEvent = newName.startsWith('-')
if (isInternalEvent) {
  // It kinda ok I guess, so only a log message here.
  deprecate.log(...)
} else {
  // The event cannot be used anymore.
  deprecate.warn(...)
}
```